packages/types/src/index.ts

```diff
@@ -36,146 +36,579 @@ export interface Dialog {
+   * the menu. Maximum length for the text in this field is 150 characters.
+   */
   placeholder?: PlainTextElement;
 }
 
+/**
+ * This select menu will load its options from an external data source,
+ * allowing for a dynamic list of options.
+ *
+ * ## Setup
+ * To use this menu type, you'll need to configure your app first:
+ *
+ * 1. Goto your app's settings page and choose the Interactive Components feature menu.
+ * 2. Add a URL to the Options Load URL under Message Menus.
+ * 3. Save changes.
+ * Each time a select menu of this type is opened or the user starts typing in
+ * the typeahead field, we'll send a request to your specified URL. Your app should
+ * return an HTTP 200 OK response, along with an application/json post body with an object
+ * containing either an options array, or an option_groups array. Here's an example response:
+ *
+ * https://api.slack.com/reference/messaging/block-elements#external-select
+ */
 export interface ExternalSelect extends Action {
   type: 'external_select';
+
+  /**
+   * A single option that exactly matches one of the options within the options
+   * or option_groups loaded from the external data source. This option will be
+   * selected when the menu initially loads.
+   */
   initial_option?: Option;
+
+  /**
+   * A plain_text only text object that defines the placeholder text shown on the
+   * menu. Maximum length for the text in this field is 150 characters.
+   */
   placeholder?: PlainTextElement;
+
+  /**
+   * When the typeahead field is used, a request will be sent on every character
+   * change. If you prefer fewer requests or more fully ideated queries, use the
+   * min_query_length attribute to tell Slack the fewest number of typed characters
+   * required before dispatch.
+   */
   min_query_length?: number;
 }
 
+/**
+ * An interactive element that inserts a button. The button can be a trigger for
+ * anything from opening a simple link to starting a complex workflow.
+ *
+ * To use interactive elements, you will need to make some changes to prepare your
+ * app. Read our [guide to enabling interactivity](https://api.slack.com/messaging/interactivity/enabling).
+ *
+ * https://api.slack.com/reference/messaging/block-elements#button
+ */
 export interface Button extends Action {
   type: 'button';
+
+  /**
+   * A text object that defines the button's text. Can only be of type: plain_text.
+   * Maximum length for the text in this field is 75 characters.
+   */
   text: PlainTextElement;
+
+  /**
+   * The value to send along with the
+   * [interaction payload](https://api.slack.com/messaging/interactivity/enabling#understanding-payloads).
+   * Maximum length for this field is 75 characters.
+   */
   value?: string;
+
+  /**
+   * The value to send along with the
+   * [interaction payload](https://api.slack.com/messaging/interactivity/enabling#understanding-payloads).
+   * Maximum length for this field is 75 characters.
+   */
   url?: string;
 }
 
+/**
```

Comment:
I know this is copied from our docs, but it's a little long for a comment in the code. I'd suggest shortening it down to something like:

An overflow menu is a compact list of options that is opened using an ellipsis ("...") button. Unlike select menus, there is no typeahead.